### DISCUSSION TITLE
Filtering on a relation

### DISCUSSION THREAD

**Discussion by selvinkuik** [Adrinalin4ik/Nestjs-Graphql-Tools#37] [Q&A] [answered]
![Screenshot 2024-02-13 21 16 57](https://github.com/Adrinalin4ik/Nestjs-Graphql-Tools/assets/3469560/6d7c2d98-0176-4c53-aeba-d6ee98c7a985)

In the above relationship, I could write a query to find content of a specific genre.

```sql
SELECT
	*
FROM
	"content"
	JOIN "contentGenre" ON "content"."id" = "contentGenre"."contentId"
WHERE
	"contentGenre"."genreId" = 99
```

If I were to have this represented with a `content` resolver, is it possible to set up a filter in this way? i.e. filter by the genre relationship

**Comment by Adrinalin4ik** [comment, marked as answer]
@selvinkuik yes, it's possible. Please read customer filters section. Custom filters allows to filter by entities from joins.

**Reply by selvinkuik** [reply]
Awesome, I've got this working. Thanks! I have another question re: Exclusions. I have the following set up where I am trying to exclude 2 fields...

```
@ObjectType('Content')
export class ContentDto {
  @Field(() => ID)
  id!: number

  @Field(() => GraphQLISODateTime)
  @FilterField({ exclude: true })
  createdAt!: Date

  @Field(() => GraphQLISODateTime)
  @FilterField({ exclude: true })
  updatedAt!: Date
}

@Resolver(() => UserDto)
export class UserResolver {
  @ResolveField(() => [ContentDto])
  async myList(
    @Filter(() => [ContentDto, FilterInputType]) filter: FilterArgs,
    @Parent() user: UserEntity,
  ): Promise<ContentDto[]> {
    // return qb
  }
}
```

However, when I run view the docs in the GraphQL playground only the last field (in this case `updatedAt`) is excluded. Have I set up the exclusions wrong?
